File: codeforces/344A - Magnets/a.py
def main():
    n = int(input())
    last = str(input())
    groups = 1
    for i in range(1,n):
        curr = str(input())
        if curr != last:
            groups+=1
            last = curr
    print(groups)

if __name__ == '__main__':
    main()

File: codeforces/344A - Magnets/a.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;
    int last;
    cin >> last;
    int groups = 1;
    for(int i = 1; i < n; ++i) {
        int curr;
        cin >> curr;
        if(last!=curr) {
            groups++;
            last = curr;
        }
    }
    cout << groups;
    return 0;
}